scheds/rust/scx_rusty/src/bpf/main.bpf.c

```diff
@@ -561,6 +564,41 @@ static u64 sched_prio_to_latency_weight(u64 prio)
 	return sched_prio_to_weight[DL_MAX_LAT_PRIO - prio - 1];
 }
 
+static void do_iowait_boost(struct task_struct *p, struct task_ctx *taskc, s32 cpu)
+{
+	if (!iowait_boost || !p || !taskc || cpu < 0)
+		return;
+	if (!(p->flags & PF_KTHREAD) || (p->flags & TASK_IDLE))
+		return;
+
+	if (BPF_CORE_READ_BITFIELD_PROBED(p, in_iowait)) {
```

Comment:
We theoretically shouldn't need to do a probed read here given that `p` should be a trusted pointer. Does the verifier complain if you just do this?
```
if (p->in_iowait) {
        // ...
}
```